### PR TITLE
Move and extend sharding and partitioning guide

### PR DESCRIPTION
**Pull request by karynzv**
## Summary of the changes / Why this is an improvement
Rewrote the sharding and partitioning guide and moved it from getting started to best practices. For the getting started section, I'd suggest having a more high-level set of pages using the default configurations provided by CrateDB.
This page contains more detailed example to guide the new users in the sharding partitioning journey.

## Checklist

 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: docs/best-practices/configuring-sharding-partitioning.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
+.. _configuring-sharding-partitioning:
+
+=====================================
+Configuring sharding and partitioning
+=====================================
```
Comment (amotl) [inline docs/best-practices/configuring-sharding-partitioning.rst:5]: Just a suggestion for a different page title, with one less "-ing" suffix.

```suggestion
.. _sharding-partitioning:

===============================
About Sharding and Partitioning
===============================
```

File: docs/best-practices/configuring-sharding-partitioning.rst [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+
+
+Introduction
+============
+Sharding and partitioning are techniques used to distribute data evenly across multiple nodes in a cluster, ensuring data scalability, availability, and performance. 
+This page provides an in-depth guide on how to configure sharding and partitioning in CrateDB, presenting best practices and examples.
```
Comment (amotl) [inline docs/best-practices/configuring-sharding-partitioning.rst:16]: Just trimming line-length here.
```suggestion
Sharding and partitioning are techniques used to distribute data evenly across
multiple nodes in a cluster, ensuring data scalability, availability, and performance.
This page provides an in-depth guide on how to configure sharding and partitioning in
CrateDB, presenting best practices and examples.
```

File: docs/best-practices/configuring-sharding-partitioning.rst [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    node to 1000. Any operation that would exceed that limit, leads to an 
+    exception.
+
+
+Partitioning
+============
+
+CrateDB also supports splitting up data across another dimension with 
+:ref:`partitioning <crate-reference:partitioned-tables>`. You can think of a
+partition as a set of shards. For each partition the number of shards defined
```
Comment (amotl) [inline docs/best-practices/configuring-sharding-partitioning.rst:78]: Just a suggestion for adding punctuation.

```suggestion
partition as a set of shards. For each partition, the number of shards defined 
```

File: docs/best-practices/configuring-sharding-partitioning.rst [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+and then querying for the total amount of shards for the table:
+
+.. code-block:: psql
+
+    SELECT COUNT(*) FROM sys.shards
+    WHERE table_name = 'second_table';
+
+We can see that the table is split into 4 shards.
+
+Adding another row to the table with a different partition key (i.e. different
```
Comment (amotl) [inline docs/best-practices/configuring-sharding-partitioning.rst:110]:
```suggestion
Adding another record to the table with a different partition key (i.e. different 
```

File: docs/best-practices/configuring-sharding-partitioning.rst [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    exception.
+
+
+Partitioning
+============
+
+CrateDB also supports splitting up data across another dimension with 
+:ref:`partitioning <crate-reference:partitioned-tables>`. You can think of a
+partition as a set of shards. For each partition the number of shards defined 
+by ``CLUSTERED INTO x SHARDS`` are created, when a first row with a specific
```
Comment (amotl) [inline docs/best-practices/configuring-sharding-partitioning.rst:79]:
```suggestion
by ``CLUSTERED INTO x SHARDS`` are created, when a first record with a specific 
```

File: docs/best-practices/configuring-sharding-partitioning.rst [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+=====================================================
+An optimal sharding and partitioning strategy always depends on the specific 
+use case and should typically be determined by conducting 
+benchmarks across various strategies. The following steps provide a general guide for a benchmark.
+
+- Identify the ingestion rate
+- Identify the record size
+- Calculate the Throughput
+
+Then, to calculate the number of shards, you should consider that each shard size should be between 5 - 100 GB and each node can only hold up to 1000 shards.
```
Comment (amotl) [inline docs/best-practices/configuring-sharding-partitioning.rst:132]: Line wrapping and a bit of wording.
```suggestion
Then, to calculate the number of shards, you should consider that the size of each
shard should roughly be between 5 - 100 GB, and that each node can only manage
up to 1000 shards.
```

File: docs/best-practices/configuring-sharding-partitioning.rst [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+Time-series example
+-------------------
+
+To illustrate the steps above, use it in an example. For instance, imagine you want to create a *partitioned table* on a *three-node cluster* 
+to store time-series data with the following assumptions:
```
Comment (amotl) [inline docs/best-practices/configuring-sharding-partitioning.rst:138]: Line wrapping and wording.
```suggestion
To illustrate the steps above, let's use them on behalf of an example. Imagine
you want to create a *partitioned table* on a *three-node cluster* to store
time-series data with the following assumptions:
```

File: docs/best-practices/configuring-sharding-partitioning.rst [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+- Inserts: 1.000 records / s
+- Record size: 128 byte / record
+- Throughput: 125 KB / s or 10.3 GB / day
+
+Given the daily throughput is around 10 GB/day, the monthly throughput is 30 times that (~ 300 GB). The partition column can be day, week, month, quarter, etc. So, assuming a 
+monthly partition, the next step is to calculate the number of shards with the **shard size restriction** 
+(5 - 100 GB) and the **number of nodes** in the cluster in mind.
+
+With three shards, each shard will hold 100 GB (300 GB / 3), which is too
+close to the upper limit. With six shards, each shard will have 50 GB (300 GB / 6) which is closer to the middle of the recommended size range (5 - 100 GB).
```
Comment (amotl) [inline docs/best-practices/configuring-sharding-partitioning.rst:149]: Line wrapping and wording.
```suggestion
With three shards, each shard will hold 100 GB (300 GB / 3), which is too
close to the upper limit. With six shards, each shard will manage 50 GB
(300 GB / 6) of data, which is closer to the recommended size range (5 - 100 GB).
```

**Review by amotl — CHANGES_REQUESTED** [review]
Dear Karyn,

thanks for this patch, I like the guiding attitude and the reading flow very much. While conducting a quick review, I've added a few suggestions. Feel free to use them or not, as you see fit.

With kind regards,
Andreas.

**Review by hammerhead — APPROVED** [review]

**Review by hlcianfagna — APPROVED** [review]

**Review by amotl — APPROVED** [review]

**Pull request merged**